content/en/docs/tasks/access-kubernetes-api/custom-resource-definition-versioning.md

```diff
@@ -0,0 +1,238 @@
+`kubectl create` to create it.
+
+```shell
+kubectl create -f my-versioned-crontab.yaml
+```
+
+After creation, the API server starts to serve each enabled version at an HTTP
+REST endpoint. In the above example, the APIs are available at
+`/apis/example.com/v1beta1` and `/apis/example.com/v1`.
+
+### Version order
+
+The versions in a CustomResourceDefinition are automatically prioritized in the
+following way, regardless of the order in which they are defined. The version
+that comes first in term of priority is used by kubectl as the default version
+to access objects. 
+
+The version _name_ in the CustomResourceDefiniton version list is used to
+compute the order of versions, using the following algorithm to attempt to parse
+the versions by version, then stability level. 
+
+Versions of the Kubernetes versioning scheme start with a `v` followed by a
+number, `vbeta` followed by a number, or `vbeta` followed by a number. The
+Kubernetes community calls this scheme "GA versions". Within GA versions, the
+following rules apply:
+  
+- `v` is sorted before `vbeta`, which is sorted before `valpha`.
+- Numbers are sorted largest to smallest.
+- If the strings `beta`, `alpha` follow the number, they sorted in
+  that order, after the equivalent string without the `beta` or `alpha` suffix.
+- If another number follows the `beta`, or `alpha`, those numbers
+  are sorted from largest to smallest.
+- Strings that don't fit the above format are sorted alphabetically.
+
+This might make sense if you look at the following sorted version list:
+
+```none
+- v10
+- v2
+- v1
+- v11beta2
+- v10beta3
+- v3beta1
+- v12alpha1
+- v11alpha2
+- foo1
+- foo10.
+```
+
+For the example in [Specify multiple versions](#specify-multiple-versions), the
+version sort order is `v1`, followed by `v1beta`. This causes the kubectl
+command to use `v1` as the default version unless the provided object specifies
+the version.
+
+## Manage object versions
+
+An object is stored in etcd with the storage version at the time of creation.
+When the object is requested, the API server returns it in the API version of
+the request. In the case of CREATE or UPDATE requests it persists the object
+with the current storage version, not the request version.
+
+To illustrate how this works, you can save the following YAML to
+`my-crontab-v1.yaml`:
+
+```yaml
+apiVersion: "example.com/v1beta1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+spec:
+  cronSpec: "* * * * */5"
+  image: my-awesome-cron-image
+```
+
+You can create the `v1beta1` version using the following `kubectl` command.
+
+```shell
+kubectl create -f my-crontab.yaml
+```
+
+The new object is stored as `example.com/v1beta1` in etcd because `v1beta1` is
+the current storage version. If you add a definition for `v1`, it will now be
+first in the sort order, and a newly-created object that doesn't specify an API
+version will be stored as `example.com/v1`.
```

Comment:
This is not accurate. Just adding a definition for `v1` will not make it storage version even if it has higher priority. There is a `storage` flag need to be set to change storage version.

Reply:
The example bellow shows the request version is `v1` as it is preferred version and API server change the version of object to the request version (storage version is still unmodified).